src/main/java/com/mraof/minestuck/block/CassettePlayerBlock.java

```diff
@@ -0,0 +1,129 @@
+package com.mraof.minestuck.block;
+
+import com.mraof.minestuck.item.CassetteItem;
+import com.mraof.minestuck.tileentity.CassettePlayerTileEntity;
+import com.mraof.minestuck.util.CustomVoxelShape;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.item.ItemEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.state.BooleanProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityType;
+import net.minecraft.util.Hand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+import java.util.function.Supplier;
+
+
+public class CassettePlayerBlock extends DecorBlock {
+    public static final BooleanProperty HAS_CASSETTE = ModBlockStateProperties.HAS_CASSETTE;
+    public static final BooleanProperty IS_OPEN = ModBlockStateProperties.IS_OPEN;
+    private final Supplier<TileEntityType<?>> tileType;
```

Comment:
This way it would be easier when using the cassette player with alchimy. I don't plan to create more block than this one but other devs may want to.

Reply:
Even then, not much functionality is handled by the tile entity at this time, so it would make sense to at least imagine a scenario when you'd want a different supplier/tile entity. Also, the purpose of providing a tile entity type instead of a tile entity hasn't been made clear (my first point), so I'd suggest at least changing that.